src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -104,64 +108,66 @@ applyCmd logger pactDbEnv minerInfo gasModel pd spv cmd = do
 
     let buyGasEnv = CommandEnv Nothing Transactional pactDbEnv logger freeGasEnv pd' spv
 
-    buyGasResultE <- catchesPactError $! buyGas buyGasEnv cmd minerInfo supply
+    buyGasResultE <- catchesPactError $! buyGas buyGasEnv cmd minerInfo supply mcache
 
     case buyGasResultE of
 
       Left e1 ->
-        jsonErrorResult buyGasEnv requestKey e1 [] (Gas 0)
+        jsonErrorResult buyGasEnv requestKey e1 [] (Gas 0) mcache
           "tx failure for requestKey when buying gas"
-
-      Right buyGasResult -> do
-
+      Right (Left e) ->
```

Comment:
I... sense Monads...

Here and below with the `Right (Right ...`

Reply:
Here, we must fail explicitly on left with `internalError` as opposed to returning a command result, so we must case.